nneighbours.m

```diff
@@ -1,36 +1,84 @@
-function c = nneighbours(c, clip)
-    % Image origin is top left. We want bottom left so invert y
-    x = figure;plot(c(:, 4), c(:, 1), 'k.'); hold on;
-    y = figure;plot(c(:, 4), size(clip,1)-c(:, 2), 'k.'); hold on;
-    c = [c zeros(size(c,1), 1)];
-    group = 1; % Connected points share a same group id.
-    maxdist = 2600; % Max neighbour distance between frames.
+function cout = nneighbours(candidates)
+    c = candidates(candidates(:,3)==1, :);
+    c(:, end+1) = 0;
+    cout = zeros(0, size(c,2));
+    group = 1; % Group id.
+    maxdist = 140; % Max length of valid first segments.
+    maxerror = 100; % Max error between prediction and actual
+    maxguess = 4; % Max # guess in a row terminates curve
+    straightness = 0.5; % Panalize best fit curvature. I got some...
+                        % interesting results without this.
     
-    for p=1:size(c, 1)
-        cp = c(p, :); 
-        targets = find(ismember(c(:,4), cp(4)+1:cp(4)+2)); % All entries from following 2 frames
-        for tidx=transpose(targets)
-            tp = c(tidx, :);
-            if (cp(1)-tp(1))^2 + (cp(2)-tp(2))^2 < maxdist
-                % Plot segment
-                segment = [cp; tp];
-                figure(x); plot(segment(:,4), segment(:,1), 'k');
-                figure(y); plot(segment(:,4), size(clip,1)-segment(:,2), 'k');
-                
-                % Assign group id
-                if cp(5) ~= 0 
-                    c(tidx, 5) = cp(5);
-                else
-                    c(p, 5) = group;
-                    c(tidx, 5) = group;
-                    group = group +1;
-                end
-                break; % We want the nearest one so stop
+    for i=1:size(c,1)
+        cp = c(i, :); 
+        if cp(5) ~= 0  % Inner node of some curve is unlikely another start

```

Comment:
Then the first two points in first frame should both find this point in second frame. The challenge is the reverse.